gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -113,12 +118,18 @@ QueryRequest buildQueryRequest(
         attributeMetadataProvider.getAttributesMetadata(requestContext, request.getContext());
     if (hasOnlyAttributeSource(request.getFilter(), AttributeSource.EDS, attributeMetadataMap)) {
       entityIds = getEntityIdsToFilterFromSourceEDS(requestContext, request, attributeMetadataMap);
+
+      if (entityIds.isEmpty()) {
```

Comment:
nit - worth a comment explaining why we shouldn't build a query request in this case